From 712935b37bd58f8c18dd7cc02bbdf464b6accb7f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:43 -0400
Subject: [usb] uas: Fix task-management not working when connected over USB-2

Message-id: <1410887463-20674-332-git-send-email-dzickus@redhat.com>
Patchwork-id: 95366
O-Subject: [RHEL7 PATCH 331/591] uas: Fix task-management not working when connected over USB-2
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit f323abcda35ea4bae851c9be8f115ee45cc9cf22
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Tue Nov 12 10:51:33 2013 +0100

    uas: Fix task-management not working when connected over USB-2

    For USB-2 connections the stream-id must always be 0.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index 019f203..10e580e 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -746,7 +746,8 @@ static int uas_eh_task_mgmt(struct scsi_cmnd *cmnd,
 
  devinfo->running_task = 1;
  memset(&devinfo->response, 0, sizeof(devinfo->response));
- sense_urb = uas_submit_sense_urb(shost, GFP_NOIO, tag);
+ sense_urb = uas_submit_sense_urb(shost, GFP_NOIO,
+      devinfo->use_streams ? tag : 0);
  if (!sense_urb) {
   shost_printk(KERN_INFO, shost,
         "%s: %s: submit sense urb failed\n",
-- 
1.7.1